File: GridWallManagement.App.Models/Response/Account/AuthenticateResponse.cs
using GridWallManagement.App.Models.Common;
using GridWallManagement.App.Models.Response.Role;

namespace GridWallManagement.App.Models.Response.Account
{
    public class AuthenticateResponse : BaseDTO
    {
        public string Username { get; set; }

        public string Email { get; set; }

        public string MobileNumber { get; set; }

        public string PasswordHash { get; set; }

        public string RoleId { get; set; }

        public string JwtToken { get; set; } = String.Empty;

        public string RefreshToken { get; set; } = string.Empty;

        public RolesResponse Roles { get; set; }
    }
}

File: GridWallManagement.Web/src/app/shared/models/response/account/AuthenticateResponse.ts
import { BaseModel } from '../../common/BaseModel';

export class AuthenticateResponse extends BaseModel {
  username: string;
  email: string;
  mobileNumber: string;
  passwordHash: string;
  roleId: string;
  jwtToken: string;
  refreshToken: string;

  constructor(
    username: string,
    email: string,
    mobileNumber: string,
    passwordHash: string,
    roleId: string,
    jwtToken: string = '',
    refreshToken: string = ''
  ) {
    super();
    this.username = username;
    this.email = email;
    this.mobileNumber = mobileNumber;
    this.passwordHash = passwordHash;
    this.roleId = roleId;
    this.jwtToken = jwtToken;
    this.refreshToken = refreshToken;
  }
}
